lib/arrays.md

```diff
@@ -5,44 +5,86 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is an Array?
 
-Replace me with your response...
+An array is a data object that stores an ordered list of values. 
 
 # What are some examples of information that would be Arrays as opposed to some other data type?
 
-Replace me with your response...
+Arrays can consist of any data type (even more arrays), and are accessed by index numbers. Strings are read character-by-character, in contrast. Arrays can also be used to control flow, though I am not sure just how this works.
+Any list whose contents would need to be recalled individually could be stored in an array. 
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+Retrieving an element can be done with:
+
+arr = []
+arr[0] = "Jones"
+arr[1] = "Koresh"
+arr[2] = "Rajneesh"
+arr[3] = "Applewhite"
+puts arr [2]
+
+Because Ruby is zero-based indexed, to access the 6th element, use:
+
+puts arr [5]
+
+For the 8th element,
+
+puts arr [7]
+
+If we try to retrieve an element that doesn't exist in the array, we get an empty line.
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+Yes, the -6th element in the array would be the 6th from the end, counting backwards with the last element being '1'.
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+Using my above example, I put the cult leaders' names into lowercase:
+
+arr.each{ |x| puts x.downcase}      
 
 # How do you add elements to an Array?
 
-Replace me with your response...
+Simply, we can add elements to an array using:
+
+arr.push ("Jouret"),("Shinrikyo") 
 
 # Given the Array `["Laura", "Fiona", "Tori"]`, how would you replace `"Fiona"` with `"Florence"` so that you end up with `["Laura", "Florence", "Tori"]`?
 
-Replace me with your response...
+I came up with:
+
+arr = ["Laura", "Fiona", "Tori"]
+arr.map! { |element|
+   if(element == "Fiona")
+       "Florence" 
+   else
+       element
+   end
+}
+puts arr
```

Comment:
Creative! I like the thought process, but I'd also like for you to find the more straightforward way to do this.

The task is much simpler than your answer might indicate: You're trying to change the value of an element in an array at a given position. You already know how to reference an element by its given position, so setting that equal to something will take no more than one line of code.

I'm being deliberately opaque here, so please feel free to follow up if you're still unclear after some research and thought.